modules/@apostrophecms/permission/ui/apos/components/AposPermissionGrid.vue

```diff
@@ -0,0 +1,174 @@
+              :icon-color="permission.value ? 'var(--a-success)' : 'var(--a-base-5)'"
+            />
+            <span v-if="permission.value" class="apos-sr-only">
+              {{ $t('apostrophe:enabled') }}
+            </span>
+            <span v-else class="apos-sr-only">
+              {{ $t('apostrophe:disabled') }}
+            </span>
+          </dd>
+          <dt class="apos-input__role__permission-grid__label">
+            {{ $t(permission.label) }}
+          </dt>
+        </div>
+      </dl>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'AposPermissionGrid',
+  props: {
+    apiParams: {
+      type: Object,
+      required: true,
+      validator(value) {
+        if (typeof value !== 'object') {
+          return false;
+        }
+
+        const ROLE = 'role';
+        const ROLES_BY_TYPE = 'rolesByType';
+        const GROUPS = '_groups';
+
+        const keys = Object.keys(value);
+        const [ key ] = keys;
+
+        if (keys.length > 1 || ![ ROLE, ROLES_BY_TYPE, GROUPS ].includes(key)) {
+          return false;
+        }
+        if (value[ROLE] && typeof value[ROLE] !== 'string') {
+          return false;
+        }
+        if (value[ROLES_BY_TYPE] && !Array.isArray(value[ROLES_BY_TYPE])) {
+          return false;
+        }
+        if (value[GROUPS] && !Array.isArray(value[GROUPS])) {
+          return false;
+        }
+        return true;
+      }
+    }
+  },
+  data() {
+    return {
+      permissionSets: []
+    };
+  },
+  watch: {
+    apiParams: {
+      async handler() {
+        this.permissionSets = await this.getPermissionSets();
+        console.log('WATCHED - this.permissionSets', this.permissionSets);
+      },
+      deep: true
+    }
+  },
+  async mounted() {
+    if (this.next) {
+      this.permissionSets = await this.getPermissionSets();
+      console.log('MOUNTED - this.permissionSets', this.permissionSets);
+    }
+  },
+  methods: {
+    getTooltip(includes) {
+      const html = document.createElement('div');
+      html.setAttribute('class', 'apos-info');
+      const list = document.createElement('ul');
+      const intro = document.createElement('p');
+      const followUp = document.createElement('p');
+      intro.appendChild(document.createTextNode(this.$t('apostrophe:piecePermissionsIntro')));
+      followUp.appendChild(document.createTextNode(this.$t('apostrophe:piecePermissionsPieceTypeList')));
+      html.appendChild(intro);
+      html.appendChild(followUp);
+      includes.forEach(item => {
+        const li = document.createElement('li');
+        li.appendChild(document.createTextNode(this.$t(item)));
+        list.appendChild(li);
+      });
+      html.appendChild(list);
+
+      return {
+        content: html,
+        localize: false
+      };
+    },
+    async getPermissionSets() {
+      console.log('this.apiParams', this.apiParams);
+      const { permissionSets } = await apos.http.post(`${apos.permission.action}/grid`, {
```

Comment:
❓ should we catch the error and log it in the browser console?
It's already in the server console (see screenshot below), but we could catch the error properly and idisplay a user-friendly message like "an error has occurred", wdyt?

![image](https://user-images.githubusercontent.com/8301962/181602929-b3b4ba7d-f6c3-46b8-9747-c9f51411e0a1.png)